### ISSUE TITLE
[web] ChromeVox stuck on first item in a scrollable list

### ISSUE THREAD

**Issue by marcianx** [flutter/flutter#128709]
b/286441683

### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. See the code below. Importantly, it uses a `ListView` with large complex items and uses a large `cacheExtent` so that `SemanticsNode`s for all children are generated to demonstrate that this issue is not due to a lack of semantics in the DOM, which is what I originally attributed the issue to.
2. Resize the window small so that only half the first text area is visible (see Screencast below).
3. Use ChromeVox to navigate through this UI using basic navigation (Search + Left/Right shortcuts).

### Expected results

ChromeVox should be able to navigate through all items in the view.

### Actual results

ChromeVox keeps looping through various parts of the first item in the `ListView` (see screencast).

### Code sample

<details><summary>Code sample</summary>

It approximates some parts of the apps structure where I demonstrated this. It includes some additional code for debugging to be able to visually confirm that the semantics tree is fine.

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const ScrollBug());
  SemanticsBinding.instance.ensureSemantics();
}

class ScrollBug extends StatelessWidget {
  const ScrollBug({super.key});

  @override
  Widget build(BuildContext context) {
    WidgetsBinding.instance.addPostFrameCallback((_) {
      debugDumpSemanticsTree();
    });
    return MaterialApp(
      title: 'Scroll bug',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: Scaffold(
        appBar: AppBar(
          backgroundColor: Theme.of(context).colorScheme.inversePrimary,
          title: Text('Scroll bug'),
        ),
        body: ListView(
          cacheExtent: 1000,  // So that semantics for all 3 children is generated.
          children: <Widget>[
            _longContent(context, 1),
            _longContent(context, 2),
            _longContent(context, 3),
          ],
        ),
      ),
    );
  }

  Widget _longContent(BuildContext context, int index) {
    final message1 = 'The quick brown fox jumped over the lazy dog $index.';
    final message2 = 'The five boxing wizards jump quickly $index.';
    return Column(
      children: [
        LayoutBuilder(builder: (context, constraints) {
          return Row(
            children: [
              SizedBox(
                width: constraints.maxWidth / 2,
                child: outlined(RichText(
                    text: TextSpan(
                        text: '$message1 $message1 $message1 $message1'))),
              ),
              SizedBox(
                width: constraints.maxWidth / 2,
                child: outlined(RichText(
                    text: TextSpan(
                        text: '$message2 $message2 $message2 $message2'))),
              ),
            ],
          );
        }),
        outlined(Semantics( // Needed due to go/flutter-issues/26336
          container: true,
          child: const TextField(maxLines: null, minLines: 8),
        )),
        outlined(Wrap(children: [
          TextButton(onPressed: () {}, child: Text('Testing $index'))
        ])),
      ],
    );
  }
}

Widget outlined(Widget child) {
  return Padding(
    padding: const EdgeInsets.all(4),
    child: Container(
      decoration: BoxDecoration(border: Border.all()),
      padding: const EdgeInsets.all(10),
      child: child,
    ),
  );
}
```
</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[ChromeVox stuck on first item.webm](https://github.com/flutter/flutter/assets/370338/53de3855-712a-4d39-bb1c-312078a4572b)

</details>

### Logs

N/A

### Flutter Doctor output

<details><summary>Doctor output</summary>

I had to run a server within google3 to be able to access it remotely on my ChromeBook. So this is the output of `fg3 doctor -v`.  I've edited the paths out with `[...]`.

```console
[   +6 ms] ProcessManager.runSync('which java')
[ +374 ms] ProcessManager.runSync('which java')
[  +14 ms] java -version
[   +3 ms] ProcessManager.run('/usr/local/buildtools/java/jdk/bin/java -version')
[   +4 ms] executing: [...]/java/android/android_sdk_linux/platform-tools/adb.static devices -l
[   +4 ms] ProcessManager.run('[...]/java/android/android_sdk_linux/platform-tools/adb.static devices -l')
[   +3 ms] Flag --fuchsia not found, Fuchsia device discovery disabled.
[   +3 ms] Gallium device polling: connecting to galliumd.
[   +5 ms] [✓] Flutter (Channel google3, on Debian GNU/Linux rodete 6.1.25-1rodete1-amd64, locale en_US.UTF-8)
[   +3 ms]     • Framework revision 9e4104b094 (19 days ago), 2023-05-24T00:00:00.000
[   +3 ms]     • Engine revision 62a83490ee
[   +4 ms]     • Dart version 96e29af6e0
[ +338 ms] [✓] Android toolchain - develop for Android devices (Android SDK version Stable)
[   +4 ms]     • Android SDK at google3
[   +3 ms]     • Platform Stable, build-tools Stable
[   +3 ms]     • Java binary at: /usr/local/buildtools/java/jdk/bin/java
[   +3 ms]     • Java version OpenJDK Runtime Environment (build 11.0.16.1+1-google-release-533553428)
[   +4 ms] [!] Android Studio (not installed)
[   +3 ms]     • Android Studio not found; download from https://developer.android.com/studio/index.html
[   +3 ms]       (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).
[   +4 ms] [✓] VS Code (version 1.78.0)
[   +3 ms]     • VS Code at /usr/share/code
[   +4 ms]     • Flutter extension can be installed from:
[   +3 ms]       🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[   +5 ms] Gallium device polling failed to connect to galliumd: gRPC Error (code: 14, codeName: UNAVAILABLE, message: Error connecting: SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = localhost, port = 33174, details: null, rawResponse:
null, trailers: {})
[  +18 ms] List of devices attached


           * daemon not running; starting now at tcp:5037
           * daemon started successfully
[   +4 ms] executing: uname -m
[   +3 ms] ProcessManager.runSync('uname -m')
[   +9 ms] Exit code 0 from: uname -m
[   +3 ms] x86_64
[   +4 ms] executing: [...]/java/android/android_sdk_linux/platform-tools/adb.static devices -l
[   +3 ms] ProcessManager.run('[...]/java/android/android_sdk_linux/platform-tools/adb.static devices -l')
[  +11 ms] List of devices attached
[   +4 ms] [✓] Connected device (1 available)
[   +3 ms]     • Linux (desktop) • linux • linux-x64 • Debian GNU/Linux rodete 6.1.25-1rodete1-amd64
[   +4 ms] [✓] Google3 (on linux)
[   +3 ms]     • KVM enabled
[   +3 ms] ! Doctor found issues in 1 category.
[   +3 ms] "flutter doctor" took 899ms.
[+2858 ms] ensureAnalyticsSent: 2853ms
[   +4 ms] Running 0 shutdown hooks
[   +4 ms] Shutdown hooks complete
[   +4 ms] exiting with code 0
```
</details>

**Comment by darshankawar** [comment]
Thanks for the report. This seems to be specific to chromebook + chromevox, which I don't have to verify myself, so based on the report, labeling it for tracking and team's attention.

**Comment by chunhtai** [comment]
Can't reproduce with macos chrome, will need to get a chromevox devices

**Comment by chunhtai** [comment]
bumping up to p1 since this essentially makes customers product unusable in chromevox

**Comment by chunhtai** [comment]
Can reproduce it on chrome book, but not every time. It seems like there is a chance it will stuck

**Comment by chunhtai** [comment]
looks like this only happens when changing moving focus to next item that requires a scroll down. If the previous focused textfield still present after the scroll, the chromevox seems to fire another focus event to the previous focused. Needs to investigate why.

**Comment by chunhtai** [comment]
Some notes:

1. having unique id for entire tree doesn't affect the result
2. if the textfield is disabled, I no longer see the refocus behavior if i move the a11y focus out of the text field.

I think the issue then may be that the active textfield may be sending some data to html that causes this, or some div flag reflecting the active textfield is not update correctly, or timing of the above operations do not meet chromevox's requirements. Will need to investigate more

**Comment by chunhtai** [comment]
looks like this line causes the refocus https://github.com/flutter/engine/blob/3b00379252999b75e2909466888b589c3e99f3ca/lib/web_ui/lib/src/engine/semantics/text_field.dart#L425

will put up a fix soon
